apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -633,34 +622,25 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     end
   end
 
-  def register_device_with_interface(db_client, device_id, interface_name, interface_major) do
-    key_insert_statement = """
-    INSERT INTO kv_store (group, key)
-    VALUES (:group, :key)
-    """
-
-    major_str = "v#{Integer.to_string(interface_major)}"
+  def register_device_with_interface(realm, device_id, interface_name, interface_major) do
+    keyspace_name = Realm.keyspace_name(realm)
     encoded_device_id = CoreDevice.encode_device_id(device_id)
 
-    insert_device_by_interface_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(key_insert_statement)
-      |> DatabaseQuery.put(:group, "devices-by-interface-#{interface_name}-#{major_str}")
-      |> DatabaseQuery.put(:key, encoded_device_id)
-      |> DatabaseQuery.consistency(:each_quorum)
-
-    insert_to_with_data_on_interface =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(key_insert_statement)
-      |> DatabaseQuery.put(
-        :group,
-        "devices-with-data-on-interface-#{interface_name}-#{major_str}"
-      )
-      |> DatabaseQuery.put(:key, encoded_device_id)
-      |> DatabaseQuery.consistency(:each_quorum)
+    devices_by_interface = %{
+      "group" => "devices-by-interface-#{interface_name}-v#{interface_major}",
+      "key" => encoded_device_id
+    }
 
-    with {:ok, _result} <- DatabaseQuery.call(db_client, insert_device_by_interface_query),
-         {:ok, _result} <- DatabaseQuery.call(db_client, insert_to_with_data_on_interface) do
+    devices_on_interface = %{
+      "group" => "devices-with-data-on-interface-#{interface_name}-v#{interface_major}",
+      "key" => encoded_device_id
+    }
+
+    opts = [prefix: keyspace_name, consistency: :each_quorum]
+
+    # TODO: what about a batch here?
+    with {:ok, _} <- Repo.safe_insert_all(KvStore, devices_by_interface, opts),
+         {:ok, _} <- Repo.safe_insert_all(KvStore, devices_on_interface, opts) do
```

Comment:
`Repo.insert_all` expects a list of maps as its second parameter. Here we are passing a single map: `devices_by_interface` or `devices_on_interface`.